R/aggregate.R

```diff
@@ -180,43 +143,380 @@ agg_climate_data <- function(year, data_raster, climate_var, daily_agg, trans =
     merged_dt[, (list_names) := lapply(list_names, function(x) {get(x) * w_area})]
   }
 
-  # Separate month and day columns
-  merged_dt[, ':=' (month = substring(date, first=1, last=8),
-                    day = substring(date, first=10))]
+  # Separate year, month, and day columns
+  merged_dt[, ':=' (year = substring(date, first = 1, last = 9),
+                    month = substring(date, first=10, last=17),
+                    day = substring(date, first=19))]
+
+  # Temporal Aggregation
+  if(agg_to == "year"){
+    sum_by_poly <- merged_dt[,  lapply(.SD, sum), by = .(poly_id, year),
+                             .SDcols = list_names]
```

Comment:
this says you're applying `sum` across grid cells within a polygon, right? it should be a weighted average so is it just that the values for each grid cell have already been multiplied by weights?

Reply:
Yes I believe so.

Reply:
Yep all the weighted averages have already been calculated at the grid cell level and then this final aggregation is summing over the polygon by day, month, year or whatever the user chooses.